src/server/debug.rs

```diff
@@ -665,60 +669,93 @@ impl Debugger {
+        if capacity.is_err() {
+            return Err(Error::InvalidArgument(format!(
+                "bad block cache size: {:?}",
+                capacity.unwrap_err()
+            )));
+        }
+        let cache_size = capacity.unwrap().0;
+        box_try!(opt.set_block_cache_capacity(cache_size));
+        CONFIG_ROCKSDB_GAUGE
+            .with_label_values(&[cf_name, "block_cache_size"])
+            .set(cache_size as f64);
+        Ok(())
+    }
+
     pub fn modify_tikv_config(
         &self,
         module: MODULE,
         config_name: &str,
         config_value: &str,
     ) -> Result<()> {
         use super::CONFIG_ROCKSDB_GAUGE;
-        let db = match module {
-            MODULE::KVDB => DBType::KV,
-            MODULE::RAFTDB => DBType::RAFT,
-            _ => return Err(Error::NotFound(format!("unsupported module: {:?}", module))),
-        };
-        let rocksdb = self.get_db_from_type(db)?;
-        let vec: Vec<&str> = config_name.split('.').collect();
-        if vec.len() == 1 {
-            box_try!(rocksdb.set_db_options(&[(config_name, config_value)]));
-        } else if vec.len() == 2 {
-            let cf = vec[0];
-            let config_name = vec[1];
-            validate_db_and_cf(db, cf)?;
-
-            let handle = box_try!(get_cf_handle(rocksdb, cf));
-            // currently we can't modify block_cache_size via set_options_cf
-            if config_name == "block_cache_size" {
-                let opt = rocksdb.get_options_cf(handle);
-                let capacity = ReadableSize::from_str(config_value);
-                if capacity.is_err() {
+        return match module {
+            MODULE::STORAGE => {
+                if config_name != "block_cache.capacity" {
                     return Err(Error::InvalidArgument(format!(
-                        "bad block cache size: {:?}",
-                        capacity.unwrap_err()
+                        "bad argument: {}",
+                        config_name
                     )));
                 }
-                let cache_size = capacity.unwrap().0;
-                box_try!(opt.set_block_cache_capacity(cache_size));
-                CONFIG_ROCKSDB_GAUGE
-                    .with_label_values(&[cf, config_name])
-                    .set(cache_size as f64);
-            } else {
-                let mut opt = Vec::new();
-                opt.push((config_name, config_value));
-                box_try!(rocksdb.set_options_cf(handle, &opt));
-                if let Ok(v) = config_value.parse::<f64>() {
-                    CONFIG_ROCKSDB_GAUGE
-                        .with_label_values(&[cf, config_name])
-                        .set(v);
+                if !self.shared_block_cache {
+                    return Err(Error::InvalidArgument(format!(
+                        "shared block cache is disabled"
+                    )));
                 }
+                self.modify_block_cache_size(DBType::KV, CF_DEFAULT, config_value)
```

Comment:
I'm surprised passing a KV type and default CF here... 
For shared cache size, all cfs use the default CF cache?

Reply:
Since all CFs use the same cache, setting cache size to any of them will change the shared cache size. I should add some comments here. Or I can hold a `Cache` reference somewhere and use it to set the shared cache size, but that's more work. What do you think?

Reply:
I prefer more comments here.

Reply:
done